3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/pii/PiiPseudonymizerEnrichment.scala

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2017-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics
+package snowplow.enrich
+package common.enrichments.registry
+package pii
+
+// Scala
+import scala.collection.JavaConverters._
+import scala.collection.mutable.MutableList
+
+// Scala libraries
+import org.json4s
+import org.json4s.{DefaultFormats, JValue}
+import org.json4s.JsonAST._
+import org.json4s.jackson.JsonMethods
+import org.json4s.jackson.JsonMethods.{compact, parse, render}
+import org.json4s.jackson.Serialization.write
+
+// Java
+import org.apache.commons.codec.digest.DigestUtils
+
+// Java libraries
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode, TextNode}
+import com.jayway.jsonpath.{Configuration, JsonPath => JJsonPath}
+import com.jayway.jsonpath.MapFunction
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Iglu
+import iglu.client.validation.ProcessingMessageMethods._
+import iglu.client.{SchemaCriterion, SchemaKey}
+
+// This project
+import common.ValidatedNelMessage
+import common.utils.ScalazJson4sUtils
+import common.outputs.EnrichedEvent
+
+/**
+ * Companion object. Lets us create a PiiPseudonymizerEnrichment
+ * from a JValue.
+ */
+object PiiPseudonymizerEnrichment extends ParseableEnrichment {
+
+  implicit val json4sFormats = DefaultFormats
+
+  override val supportedSchema =
+    SchemaCriterion("com.snowplowanalytics.snowplow.enrichments", "pii_enrichment_config", "jsonschema", 2, 0, 0)
+
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[PiiPseudonymizerEnrichment] = {
+    for {
+      conf <- matchesSchema(config, schemaKey)
+      enabled = ScalazJson4sUtils.extract[Boolean](conf, "enabled").toOption.getOrElse(false)
```

Comment:
Can you please clarify how `enabled` works here?

As far as I can tell, `EnrichmentRegistry.buildEnrichmentConfig` checks this value before and if it is `false` then `PiiPseudonymizerEnrichment.parse` will never be invoked. Am I missing something? Also it looks strange to me that on `enabled = false` below, we still have an enrichment object, which is just no-op.